server/src/shared/infra/http/controllers/users.controller.ts

```diff
@@ -75,4 +84,25 @@ export class UsersController {
   delete(@Param('id') id: string): Promise<void> {
     return this.deleteUser.execute(id);
   }
+
+  @ApiOkResponse({
+    status: HttpStatus.OK,
+    description: 'Senha alterada com sucesso',
+  })
+  @ApiBadRequestResponse({
+    status: HttpStatus.BAD_REQUEST,
+    description: 'Não foi possível alterar a senha',
+  })
+  @Put('/changePassword')
+  UpdatePassword(
+    @Body()
+    { id, password, new_password, confirm_new_password }: UpdatePasswordDTO,
+  ): Promise<User> {
```

Comment:
O certo não seria nem enviar um usuário como resposta

https://www.moesif.com/blog/technical/api-design/Which-HTTP-Status-Code-To-Use-For-Every-CRUD-App/
https://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete